### DISCUSSION TITLE
Separator

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#236] [Component suggestions]
<img width="480" alt="visuele weergave van het separator component" src="https://github.com/nl-design-system/backlog/assets/545207/37aa7256-cc9b-433c-b419-79d0ce0a957f">

We willen graag meer leren over het Separator component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1711554472566429)

---

## Naam
Separator

## Doel
Markering van de plek waar tekst op een nieuwe manier verder gaat, zoals een nieuwe scene of een ander onderwerp.

## Ook bekend als
Divider, Horizonal rule, Thematic break.

## Onderzoek
[`<hr>`: The Thematic Break (Horizontal Rule) element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr)
Documentatie van Mozilla Developers Network over het `<hr>` HTML element.

[Not Your Typical Horizontal Rules](https://www.sarasoueidan.com/blog/horizontal-rules/)
In dit artikel beschrijft Sara Soueidan de Horizontal Rule aan de hand van semantiek, toegankelijkheid en stijling.

**Comment by MariekeBrouwer** [comment]
![image](https://github.com/nl-design-system/backlog/assets/149144374/9d7aa626-da90-4eb5-8c6f-a4511704bc4f) 
![image](https://github.com/nl-design-system/backlog/assets/149144374/78577226-9f55-4248-b0ee-9a14abfe5f53)

wij gebruiken het bijvoorbeeld op pagina's als deze om verschillende 'typen' content van elkaar te scheiden.

Belangrijk om te noemen is dat bij website pagina's er wel wordt gekeken naar de content opzet als geheel. Bij maar 1 blok content is het bijvoorbeeld niet per se logisch om de divider te gebruiken.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-data-display-divider--docs)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1596-83&mode=design&t=nKdvzE1083zR4Qm2-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=843-1112&mode=design&t=ziOXYciyk0b1yFXn-4)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Divider)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-03-28 om 12 56 44](https://github.com/nl-design-system/backlog/assets/545207/1c7e7479-2292-4ab1-9a72-3d6148a99a02)

Vertical
![Scherm­afbeelding 2024-03-28 om 12 56 48](https://github.com/nl-design-system/backlog/assets/545207/7f4e447e-8830-4568-8cd5-e98b3afdf984)

Vertical Images
![Scherm­afbeelding 2024-03-28 om 12 56 52](https://github.com/nl-design-system/backlog/assets/545207/7dd44883-d0a5-4147-9629-24dfccf6fc6b)

**Figma - huidig**

Inset
![Scherm­afbeelding 2024-03-28 om 12 58 27](https://github.com/nl-design-system/backlog/assets/545207/ac671406-4528-4b27-8ae8-4f33d33c5ac4)

Full-bleed
![Scherm­afbeelding 2024-03-28 om 12 58 34](https://github.com/nl-design-system/backlog/assets/545207/d95b3c2d-5e87-4407-bff7-f22c85f53e36)

**Figma - WIP**
![Scherm­afbeelding 2024-03-28 om 12 59 36](https://github.com/nl-design-system/backlog/assets/545207/87d7aa58-a070-4ae1-978b-9bc38c5bd823)

---


### Notities
- In Storybook 'Divider' genoemd.
- Richtlijnen beschikbaar.
- Lijkt op basis van 'Figma - huidig' eerder als border ingezet te worden?

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-horizontal-rule--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/horizontal-rule)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-03-28 om 13 02 33](https://github.com/nl-design-system/backlog/assets/545207/1d38ebbb-a43c-4684-90cf-e82923ea78fc)

---

### Notities

Onder de naam 'Horizontal rule'

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-separator--docs)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=843-1112&mode=design&t=zdb9AcVVsUOHrRyC-4)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/separator)

---

### Screenshots / Varianten

**Storybook**

Default
Markup with the `<hr>` element, styling via utrecht-separator class name. For horizontal separators, `<hr>` is the preferred markup.
![Scherm­afbeelding 2024-03-28 om 13 04 59](https://github.com/nl-design-system/backlog/assets/545207/12182608-d3dd-4703-94dc-43fc44d0c076)

Separator Div Support
Markup with a `<div>` element, with additional attributes to give the correct ARIA role. Styling via utrecht-separator class name.
![Scherm­afbeelding 2024-03-28 om 13 05 03](https://github.com/nl-design-system/backlog/assets/545207/4e9ef28b-03d9-404c-98f3-7c559f297707)

**Figma**
![Scherm­afbeelding 2024-03-28 om 13 06 53](https://github.com/nl-design-system/backlog/assets/545207/a8a49982-35f7-465d-8443-7c12c100e034)

---

### Notities
Richtlijnen beschikbaar (Storybook).

**Comment by rianrietveld** [comment]
Persoonlijk vind ik dit geen component. In alle gevallen die ik in de praktijk tegenkom wordt hiervooreen border gestyled met CSS gebruikt.
Een hr wordt voorgelezen door een screenreader, maar in veel van de bovenstaande voorbeelden is het meer een visuele aanduiding, een kader, een lijn onder een kopje etc.

ik ben hier erg bang voor de verwarring tussen het gebruik van visuele elementen en echte betekenisvolle info.

Wat is echt een goede use case om een hr te gebruiken in plaast van een CSS-border?
Als dit toch als component wordt geaccepteerd en het betekenis heeft voor het begrijpen van de inhoud moet het in ieder geval voldoen aan de kleurcontast eisen van niet-tekstuele content ([WCAG 1.4.11](https://nldesignsystem.nl/wcag/1.4.11)) en echt zin hebben en ook logisch zijn in de leesvolgorde ([WCAG1.3.2](https://nldesignsystem.nl/wcag/1.3.2)).

**Comment by rianrietveld** [comment]
Na overleg met @jeffreylauwers 
Ik ben het ermee eens om de` <hr />` naar de Hall of Fame te brengen mits goed gedocumenteerd is:
- wanneer je hem wel maar ook juist niet moet gebruiken (gerelateerd aan ([WCAG1.3.2](https://nldesignsystem.nl/wcag/1.3.2)))
- dat als je hem toepast hij moet voldoen aan WCAG non-text contrast richtlijn ([WCAG 1.4.11](https://nldesignsystem.nl/wcag/1.4.11)).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Separator op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=separator)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Separator op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32)
